toml/decoder.py

```diff
@@ -326,9 +349,50 @@ def loads(s, _dict=dict, decoder=None):
     multilinestr = ""
     multibackslash = False
     pos = 0
+    comment_norm_factor = 0
     for idx, line in enumerate(s):
         if idx > 0:
             pos += len(s[idx - 1]) + 1
+
+        if len(comment_info) > 0 and comment_info[0][0] - 1 == idx:
+            # match the comment lines with their original positions
+            # in the file and handle them appropriately
+            original_line_no = comment_lines[0] - 1
+            original_line = (original.split('\n')[original_line_no])
+            recent = 'start'
+            if original_line.strip().startswith('#'):
+                recent_line_no = original_line_no
+                while recent_line_no > 0:
+                    prev_line = original.split('\n')[recent_line_no - 1]
+                    if prev_line == '':
+                        recent_line_no -= 1
+                        continue
+
+                    if prev_line.strip().startswith('#'):
+                        recent_line_no -= 1
+                        continue
+
+                    recent = prev_line.split('=')[0].strip()
+                    break
+                decoder.handle_comment(currentlevel, original_line,
+                                       'normal', original,
+                                       recent)
+
+            elif len(s[idx].split('=')) == 2:
```

Comment:
This won't capture this line of TOML:

```
x = "Testing 3 != 4" # some string with = in it followed by a comment
```

Reply:
It captures as `s[idx]` the line from with comment is removed and not original.

Reply:
This was a nice edge case.